examples/Raytracer_Benchmark/RaytraceBenchmark.cpp

```diff
@@ -0,0 +1,159 @@
+#include "ArgParser.h"
+#include "examples/Raytracer_Benchmark/Raytracer.h"
+#include "base/inc/CopCore/include/CopCore/Global.h"
+
+#ifdef VECGEOM_GDML
+#include <VecGeom/gdml/Frontend.h>
+#endif
+
+// using namespace vecgeom;
+
+// forward declarations
+int RaytraceBenchmarkCPU(cxx::RaytracerData_t &rtdata);
+
+#ifdef VECGEOM_ENABLE_CUDA
+// namespace vecgeom {
+namespace cuda {
+struct RaytracerData_t;
+} // namespace cuda
+//} // namespace vecgeom
+
+int RaytraceBenchmarkGPU(cuda::RaytracerData_t *, bool, int);
+#endif
+
+int main(int argc, char *argv[])
+{
+#ifndef VECGEOM_GDML
+  std::cout << "### VecGeom must be compiled with GDML support to run this.\n";
+  return 1;
+#endif
+#ifndef VECGEOM_USE_NAVINDEX
+  std::cout << "### VecGeom must be compiled with USE_NAVINDEX support to run this.\n";
+  return 2;
+#endif
+
+  // geometry file name and global transformation cache depth (affects size of navigation index table)
+  OPTION_STRING(gdml_name, "trackML.gdml");
+  OPTION_INT(cache_depth, 0); // 0 = full depth
+
+  // image size in pixels
+  OPTION_INT(px, 1840);
+  OPTION_INT(py, 512);
+
+  // RT model as in { kRTxray = 0, kRTspecular, kRTtransparent, kRTdiffuse };
+  OPTION_INT(model, 2);
+
+  // RT view as in { kRTVparallel = 0, kRTVperspective };
+  OPTION_INT(view, 1);
+
+  // zoom w.r.t to the default view mode
+  OPTION_DOUBLE(zoom, 3.5);
+
+  // Screen position in world coordinates
+  OPTION_DOUBLE(screenx, -5000);
+  OPTION_DOUBLE(screeny, 0);
+  OPTION_DOUBLE(screenz, 0);
+
+  // Up vector (no need to be normalized)
+  OPTION_DOUBLE(upx, 0);
+  OPTION_DOUBLE(upy, 1);
+  OPTION_DOUBLE(upz, 0);
+  vecgeom::Vector3D<double> up(upx, upy, upz);
+
+  // Light color, object color (no color per volume yet) - in RGBA chars compressed into an unsigned integer
+  OPTION_INT(bkgcol, 0xFF0000FF); // red
+  OPTION_INT(objcol, 0x0000FFFF); // blue
+  OPTION_INT(vdepth, 4);          // visible depth
+
+  OPTION_INT(on_gpu, 1);     // run on GPU
+  OPTION_INT(use_tiles, 0);  // run on GPU in tiled mode
+  OPTION_INT(block_size, 8); // run on GPU in tiled mode
+
+// Try to open the input file
+#ifdef VECGEOM_GDML
+  vecgeom::GeoManager::Instance().SetTransformationCacheDepth(cache_depth);
+  bool load = vgdml::Frontend::Load(gdml_name.c_str(), false);
+  if (!load) return 2;
+#endif
+
+  auto world = vecgeom::GeoManager::Instance().GetWorld();
+  if (!world) return 3;
+
+  RaytracerData_t rtdata;
+
+  rtdata.fScreenPos.Set(screenx, screeny, screenz);
+  rtdata.fUp.Set(upx, upy, upz);
+  rtdata.fZoom     = zoom;
+  rtdata.fModel    = (ERTmodel)model;
+  rtdata.fView     = (ERTView)view;
+  rtdata.fSize_px  = px;
+  rtdata.fSize_py  = py;
+  rtdata.fBkgColor = bkgcol;
+  rtdata.fObjColor = objcol;
+  rtdata.fVisDepth = vdepth;
+  rtdata.fMaxDepth = vecgeom::GeoManager::Instance().getMaxDepth();
+
+  Raytracer::InitializeModel(world, rtdata);
+  rtdata.Print();
+
+  auto ierr = 0;
+  if (on_gpu) {
+#ifdef VECGEOM_ENABLE_CUDA
```

Comment:
should be VECCORE_CUDA everywhere

Reply:
Realised that this feature just checks that VecGeom was compiled with CUDA support, can be kept.